x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BigArrayVectorTests.java

```diff
@@ -78,6 +84,15 @@ public void testBoolean() throws IOException {
                     assertThat(mask.mask().getBoolean(p), equalTo(values[p]));
                 }
             }
+            if (value != null) {
```

Comment:
Using a quick Stream+toSet we would be able to check all possibilities, even if the value isn't constant. As to be as complete as possible, with not much effort (?)

Same for the other similar tests, probably

Reply:
Checking.

Reply:
I wanted to go with something that felt less like re-implementing the feature and more like knowing up front how it was built.

Reply:
With that "one-liner", we get to test all `allTrue/False` methods, which are otherwise untested from what I saw.

Not a big matter, but I would reconsider doing so, maybe in another PR. As now "allTrue/False" _has to be precise_. And it would already be tested if we do some improvement in the future, like storing if there are any true/false on `add()`

Reply:
I was going to say: "this tests it 2/3 of the time which is pretty good" but then realize I have a pretty big missing step here which is when both should return `false` - which is most of that 1/3 of the time.... Ok. I'm convinced. I'll go and do it.